pkg/daemon/daemon.go

```diff
@@ -364,6 +367,22 @@ func New(
 	}
 	tracker.processManager = pm
 
+	// Initialize fsnotify watcher for sa_file change detection
+	watcher, err := fsnotify.NewWatcher()
```

Comment:
Nit: You can use the same "saFilesWatcher" name for this local var. It's a more descriptive name to use than just "watcher".